src/components/Modal/modal.tsx

```diff
@@ -0,0 +1,175 @@
+import * as React from 'react';
+import { themr, ThemedComponentClass } from 'react-css-themr';
+import { Keys } from '../../types';
+import { MODAL } from '../ThemeIdentifiers';
+import KeypressListener from '../KeypressListener';
+import Helpers from './helpers';
+import Dialog from './dialog';
+import * as baseTheme from './Modal.scss';
+
+const bodyStyle = (pading: string, overflow: string): void => {
+  const body = document.getElementsByTagName('body');
+  body[0].style.paddingRight = pading;
+  body[0].style.overflow = overflow;
+};
+
+const getModalElement = (id: string) => {
+  return {
+    modal: Helpers.getElement(`modal-${id}`),
+    dialog: Helpers.getElement(`dialog-${id}`),
+  };
+};
+
+export type Size = 'Small' | 'Medium' | 'Large' | number;
+
+export interface Ianimate {
+  in: Function,
+  out: Function,
+}
+
+export interface Itrigger {
+  body: string,
+  animate: Ianimate,
+}
+
+export interface Props {
+  className?: string,
+  classes?: string,
+  close?: boolean,
+  footer?: React.ReactNode,
+  header?: string | React.ReactNode,
```

Comment:
You don't need string, ReactNode is sufficient.  ReactNode includes ReactChild, which includes ReactText, which includes string: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e9569f1a1de869f8b2dd3dcbfe8f7a86f6909606/types/react/index.d.ts#L131.